### PR TITLE
SFB-223: Improve formatting of ttl code

### PR DESCRIPTION
**Pull request by JonasVanHoof**
## ID
 [SFB-223](https://binnenland.atlassian.net/browse/SFB-223)

 ## Description

 When using the shortcut for formatting the ttl code it does not do this for flat ttl code. In this PR we tried improving this.

 ## Type of change

 - [ ] Bug fix
 - [ ] New feature
 - [ ] Breaking change
 - [x] Other

 ## How to test

**Unformatted code**
```ttl
@prefix : <#>.@prefix form: <http://lblod.data.gift/vocabularies/forms/>.@prefix sh: <http://www.w3.org/ns/shacl#>.
@prefix displayTypes: <http://lblod.data.gift/display-types/>.@prefix nodes: <http://data.lblod.info/form-data/nodes/>.@prefix emb: <http://ember-submission-form-fields/>.nodes:24289e48-258f-4919-8c3e-5783a6acb4a4	a form:Field;	form:displayType displayTypes:defaultInput;	form:partOf nodes:d7b33768-3723-4291-a7be-3d8a7d7cdbc1;
	form:validatedBy [		a form:ValidPhoneNumber;		form:defaultCountry "BE";
		form:grouping form:MatchEvery;
		sh:order 2;		sh:path nodes:e61f56db-6346-4a61-a75e-33e091789e40;
		sh:resultMessage "Geef een geldig telefoonnummer in."@nl ],
		[
		a form:RequiredConstraint;		form:grouping form:Bag;
		sh:order 1;		sh:path nodes:e61f56db-6346-4a61-a75e-33e091789e40;
		sh:resultMessage "Dit veld is verplicht."@nl ];	sh:name "Veldnaam";
	sh:order 2;	sh:path nodes:e61f56db-6346-4a61-a75e-33e091789e40 .
nodes:d7b33768-3723-4291-a7be-3d8a7d7cdbc1
	a form:Section;	sh:name "Sectie";
	sh:order 1 .emb:source-node
	a form:Form,	form:TopLevelForm;
	form:includes nodes:24289e48-258f-4919-8c3e-5783a6acb4a4;	form:partOf nodes:d7b33768-3723-4291-a7be-3d8a7d7cdbc1 .
```

**should result in**

```ttl
@prefix : <#>.
@prefix form: <http://lblod.data.gift/vocabularies/forms/>.
@prefix sh: <http://www.w3.org/ns/shacl#>.
@prefix displayTypes: <http://lblod.data.gift/display-types/>.
@prefix nodes: <http://data.lblod.info/form-data/nodes/>.
@prefix emb: <http://ember-submission-form-fields/>.

nodes:24289e48-258f-4919-8c3e-5783a6acb4a4
	a form:Field;
	form:displayType displayTypes:defaultInput;
	form:partOf nodes:d7b33768-3723-4291-a7be-3d8a7d7cdbc1;
	form:validatedBy [
		a form:ValidPhoneNumber;
		form:defaultCountry "BE";
		form:grouping form:MatchEvery;
		sh:order 2;
		sh:path nodes:e61f56db-6346-4a61-a75e-33e091789e40;
		sh:resultMessage "Geef een geldig telefoonnummer in."@nl ],
		[
		a form:RequiredConstraint;
		form:grouping form:Bag;
		sh:order 1;
		sh:path nodes:e61f56db-6346-4a61-a75e-33e091789e40;
		sh:resultMessage "Dit veld is verplicht."@nl ];
	sh:name "Veldnaam";
	sh:order 2;
	sh:path nodes:e61f56db-6346-4a61-a75e-33e091789e40 .

nodes:d7b33768-3723-4291-a7be-3d8a7d7cdbc1
	a form:Section;
	sh:name "Sectie";
	sh:order 1 .

emb:source-node
	a form:Form,
	form:TopLevelForm;
	form:includes nodes:24289e48-258f-4919-8c3e-5783a6acb4a4;
	form:partOf nodes:d7b33768-3723-4291-a7be-3d8a7d7cdbc1 .


```
 ## Links to other PR's

 - /

### CODE REVIEW

File: app/utils/code-editor/format/format-editor-doc.js [hunk 1]
```diff
@@ -1,23 +1,228 @@
...
+);
+const predicateWithNodeValueRegex = new RegExp(
+  /(\w+):(\w+)\s+(\w+):\s*([^;\n\s]+)\s*[,;\.]/
+);
+const predicateWithStringOrNumberValueRegex = new RegExp(
+  /(\w+):(\w+)\s+("([^"]+)"|(\d+)|(true|false)|"""([^"]+)""")\s*[;,\.]/
+);
+const startOfValidationsRegex = new RegExp(/(\w+):([\w-]+)\s[\[]/);
+const lineBeforeValidationSummation = new RegExp(/.*],/);
+const endBlankNodeRegex = new RegExp(/[\]]\s*;/);
```
Comment (ZijneDwaasheid) [inline app/utils/code-editor/format/format-editor-doc.js:166]: Lets see what type of emojis we get out of this!

**Review by ZijneDwaasheid — CHANGES_REQUESTED** [review]
When a node:id ends with an a, and the next thing is form:shape. The a will be used as rdf:type
Screenshot from 2024-03-20 14-31-49
Screenshot from 2024-03-20 14-32-16

File: app/utils/code-editor/format/format-editor-doc.js (hunk 1 above)
Comment (JonasVanHoof) [reply app/utils/code-editor/format/format-editor-doc.js:166]: image

Comment (ZijneDwaasheid) [reply app/utils/code-editor/format/format-editor-doc.js:166]: somewhat disappointing

Comment (ZijneDwaasheid) [reply app/utils/code-editor/format/format-editor-doc.js:166]: expected them to be all over the place

Comment (JonasVanHoof) [reply app/utils/code-editor/format/format-editor-doc.js:166]: haha to bad 

You can maybe create new ones for these combinations ;p

Comment (JonasVanHoof) [reply app/utils/code-editor/format/format-editor-doc.js]: Should be fixed 
image

**Review by ZijneDwaasheid — APPROVED** [review]
alrighty

**Pull request merged**